### PR TITLE
fix: return bytes or bytes32 if detected

### PR DESCRIPTION
**Pull request by odyslam**
When parsing a string, it will try the following in that order:
- It will try to parse as Address
- it will see if it starts with `0x`
- If it doesn't, it will return `string`
- else, it will return a `Bytes32` if `length` == 32 or `bytes` if not (bytes32 is a fairly common bytes type)

We could try to detect for Bytes8, 16, 32 but not sure it adds to the UX.

### CODE REVIEW

File: evm/src/executor/inspector/cheatcodes/ext.rs [hunk 1]
```diff
@@ -281,13 +281,29 @@ fn remove_file(state: &mut Cheatcodes, path: impl AsRef<Path>) -> Result<Bytes,
...
+                // if length == 32 bytes, then encode as Bytes32, else Bytes
+                Ok(if arr.len() == 64 {
+                    Token::FixedBytes(Vec::from_hex(arr).unwrap())
+                } else {
+                    Token::Bytes(Vec::from_hex(arr).unwrap())
+                })
+            } else {
+                let arr = format!("0{}", val.strip_prefix("0x").unwrap());
+                Ok(Token::Bytes(Vec::from_hex(arr).unwrap()))
+            }
```
Comment (mattsse) [inline evm/src/executor/inspector/cheatcodes/ext.rs:306]: there's `str::strip_prefix` which returns an option, so this should be an `if let Some() val.strip_prefix("0x")`

File: evm/src/executor/inspector/cheatcodes/ext.rs [hunk 2]
```diff
@@ -281,13 +281,29 @@ fn remove_file(state: &mut Cheatcodes, path: impl AsRef<Path>) -> Result<Bytes,
...
+            println!("val {val}");
+            // If incorrect length, pad 0 at the beginning
+            if val.len() % 2 == 0 {
+                let arr = val.strip_prefix("0x").unwrap();
+                // if length == 32 bytes, then encode as Bytes32, else Bytes
+                Ok(if arr.len() == 64 {
+                    Token::FixedBytes(Vec::from_hex(arr).unwrap())
+                } else {
+                    Token::Bytes(Vec::from_hex(arr).unwrap())
+                })
```
Comment (mattsse) [inline evm/src/executor/inspector/cheatcodes/ext.rs:302]: can try: `if let Ok(bytes) = hex::decode()`

File: evm/src/executor/inspector/cheatcodes/ext.rs [hunk 3]
```diff
@@ -281,13 +281,29 @@ fn remove_file(state: &mut Cheatcodes, path: impl AsRef<Path>) -> Result<Bytes,
 /// it will call itself to convert each of it's value and encode the whole as a
 /// Tuple
 fn value_to_token(value: &Value) -> Result<Token, Token> {
+    println!("value to token: {:?}", value);
     if value.is_boolean() {
         Ok(Token::Bool(value.as_bool().unwrap()))
     } else if value.is_string() {
         let val = value.as_str().unwrap();
```
Comment (mattsse) [inline evm/src/executor/inspector/cheatcodes/ext.rs:288]: these also could be `if let Some()` `as_string` and `as_boolean`, I believe these exist on `Value`

**Comment by odyslam** [comment]
Thanks for the comments @mattsse ! It's much cleaner now :)

**Comment by odyslam** [comment]
@mattsse let's merge this

**Review by mattsse — APPROVED** [review]
yep, lgtm

**Pull request merged**